### PR TITLE
docs: Storybook in atlantis

### PR DESCRIPTION
**Pull request by darryltec**
## Motivations

We want to move Docz documentation to Storybook. This set us up to migrate documentation over.

Keep in mind that Docz still exists and will be the one you'll see on the preview deploy until we move all the existing one to Storybook.

### Added

- Installed storybook for react
- Added mdx linters
- Added `.stories.mdx` for Button (kept the original .mdx file since Docz need that)

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

- Pull down this branch
- run `npm ci`
- run `npm run storybook`
- You should see the storybook runs with Button in it
- run `npm start` to ensure Docz still runs

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

JOB-61681

### CODE REVIEW

**Comment by chris-at-jobber** [comment]
@darryltec how do we view the Storybook instance in the browser? Is it running on a new port / is it accessible on the Cloudflare build?

**Comment by darryltec** [comment]
@chris-at-jobber not accessible yet, sadly :( Might have to ask @kingstonfung to put it up for us in cloudflare

**Comment by chris-at-jobber** [comment]
OK, local only for now. Gotcha

File: package.json [hunk 1]
```diff
@@ -55,6 +57,15 @@
     "@jobber/generators": "file:packages/generators",
     "@jobber/stylelint-config": "file:packages/stylelint-config",
     "@reach/router": "^1.3.3",
+    "@storybook/addon-actions": "^6.5.16",
```
Comment (chris-at-jobber) [inline package.json:60]: Is there a reason we aren't just starting with `7.0` since we're creating a fresh instance?

Comment (darryltec) [reply package.json:60]: I used `storybook init` on this one and that's what it installed. haha. Probably because 7 is still on beta?

Comment (chris-at-jobber) [reply package.json:60]: Ah I see `6.5.16` is the "latest" release indeed. The way they have been promoting 7 I assumed it was full release.

Comment (darryltec) [reply package.json:60]: looks pretty straightforward to go there. lemme try it. 

Yeah like this one https://nx.dev/packages/storybook/documents/storybook-7-setup It's ready!! But it's actually not!!

**Comment by chris-at-jobber** [comment]
We touched on this in a previous conversation but do we have any control over Storybook's URL structure? I would love to get away from this query + hyphen-based approach so we can keep the URLs 1:1 when we migrate.

image

**Comment by chris-at-jobber** [comment]
Do we _want_ to keep the MDX linters? What are they doing for us?

**Comment by chris-at-jobber** [comment]
One last one - I see the viewmode is set to `docs` but I'm consistently seeing `canvas` as the default.

**Comment by darryltec** [comment]
> We touched on this in a previous conversation but do we have any control over Storybook's URL structure? I would love to get away from this query + hyphen-based approach so we can keep the URLs 1:1 when we migrate.
> 
> image

still looking into this. looks like there's a way but haven't tried it yet

**Comment by darryltec** [comment]
> Do we _want_ to keep the MDX linters? What are they doing for us?

mostly autofixes as you save instead of on commit. Also tells you which imports are broken

**Comment by darryltec** [comment]
> One last one - I see the viewmode is set to `docs` but I'm consistently seeing `canvas` as the default.
> 
> EDIT: This looks like a weird nav quirk, when I navigate to `Button` in the sidenav it works as expected but when I go "home" I see the Canvas view of Button.

yeah, that confused me earlier. It's because theres only button and storybook is set up to load the first thing, it loads button in canvas mode but the moment you click the button in the sidebar, it goes to docs. It'll be fixed once we add the front page.

File: package.json [hunk 2]
```diff
@@ -30,7 +30,9 @@
     "preventManualRelease": "node scripts/preventManualRelease.js",
     "buildHooksIndex": "node scripts/createIndexFile.js ./packages/hooks/src",
     "lerna:clean": "lerna run clean && lerna clean --y && npm run clean:packages && npm run clean",
-    "clean:packages": "rm -rf node_modules **/dist"
+    "clean:packages": "rm -rf node_modules **/dist",
+    "storybook": "start-storybook -p 6006",
```
Comment (rodrigoeidelvein) [inline package.json:34]: Maybe we could change the port to not conflict with jobber-mobile's storybook so we can have both instances running?

Comment (darryltec) [reply package.json:34]: oh that's a good idea. `6009`? 😏

File: package.json (hunk 1 above)
Comment (darryltec) [reply package.json:60]: lil update on this. We need to be on node 16 to use 7 😢

File: .storybook/theme.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { JobberStyle } from "@jobber/design/foundation.js";
+
+// eslint-disable-next-line import/no-default-export
+export default create({
+  base: "light",
+
+  /**
+   * Fonts
+   */
+  fontBase: JobberStyle["typography--fontFamily-normal"],
```
Comment (chris-at-jobber) [inline .storybook/theme.ts:11]: Do they have a concept of a display/heading font in Storybook?

Comment (darryltec) [reply .storybook/theme.ts:11]: they do not. we'd have to inject jobber components in those things. Also... i think that's the reason why the header links doesn't exist on mobile. 😆

**Review by rodrigoeidelvein — APPROVED** [review]
To start the Atlantis Storybook, it LGTM!

We can work on some overrides later, as we have in jobber-mobile.

**Comment by darryltec** [comment]
> We can work on some overrides later, as we have in jobber-mobile.

Agreed!

**Pull request merged**